src/electron/process_manager.ts

```diff
@@ -40,13 +47,19 @@ function pathToEmbeddedExe(basename: string) {
     __dirname.replace('app.asar', 'app.asar.unpacked'), 'bin', 'win32', `${basename}.exe`);
 }
 
+
+function pathToEmbeddedBinary(basename: string) {
+  return path.join(
+      __dirname.replace('app.asar', 'app.asar.unpacked'), 'bin', 'linux', `${basename}`);
+}
 // Three tools are required to launch the proxy on Windows:
 //  - ss-local.exe connects with the remote Shadowsocks server, exposing a SOCKS5 proxy
 //  - badvpn-tun2socks.exe connects the SOCKS5 proxy to a TAP-like network interface
 //  - OutlineService configures the system to route via a TAP-like network device, must be installed
 
 let ssLocal: ChildProcess | undefined;
 let tun2socks: ChildProcess | undefined;
+let tunDeviceName: string|undefined;
```

Comment:
On Windows we hard-coded the device name which simplified a number of things. Is that an option on Linux?

Reply:
In rare cases it may result in problems. @vmon correct me if I am wrong, in daemon we have provisions to change the tun device name if it already exist.
In Linux the tun device is read from the daemon, which is easy. All the communication to change routing is done through unix socket and that's where we are reading the tun device name.

Reply:
I made some changes re this. Let me know if it's ok.

Reply:
How about we make `getTunDeviceName` to return the name of the TUN device and pass it here to `startTun2socks`? Then there's no need for the `tunDeviceName`.